### DISCUSSION TITLE
(Again) How to bundle nixvim?

### DISCUSSION THREAD

**Discussion by qlibp** [nix-community/nixvim#2629] [Q&A] [answered]
Try to use `nix-portable` to bundle nixvim, but sadly, I don't know how to use it correctly... Anyone could help me out?

**Comment by traxys** [comment]
You can try https://github.com/ralismark/nix-appimage to create an appimage from the nix-derivation. I think nix-portable allows nix to be installed as non-root, not to create portable executables from nix derivations

**Reply by qlibp** [reply]
sound nice. but what the cmd looks like if i wanna to bundle nixvim via nix-appimage? I go through the nix-appimage doc, but can’t find an example using flake, which is how nixvim setup

**Reply by qlibp** [reply]
I try it out and it works! Thx. @traxys 

Another question, is cross compilation works out-ouf-box? I try the following config on my x86_64 ubuntu machine, with nix bundle command: `nix bundle --bundler github:ralismark/nix-appimage my-nixvim/`
```nix
# my flake.nix
{
  description = "A very basic flake";

  inputs.nixvim.url = "github:nix-community/nixvim";

  outputs = {
    self,
    nixvim,
    flake-parts,
  } @ inputs: let
    config = {
      colorschemes.gruvbox.enable = true;
    };
  in
    flake-parts.lib.mkFlake {inherit inputs;} {
      systems = [
        "aarch64-darwin"
        "aarch64-linux"
        "x86_64-darwin"
        "x86_64-linux"
      ];

      perSystem = {
        pkgs,
        system,
        ...
      }: let
        nixvim' = nixvim.legacyPackages.aarch64-linux; # manually change here
        nvim = nixvim'.makeNixvim config;
      in {
        packages = {
          inherit nvim;
          default = nvim;
        };
      };
    };
}

```

But got an error
```
error: a 'aarch64-linux' with features {} is required to build '/nix/store/0q91slrhl611kjhdvi1j5yrv65cik604-builder.pl.drv', 
but I am a 'x86_64-linux' with features {benchmark, big-parallel, kvm, nixos-test, uid-range}
```

**Reply by traxys** [reply]
Cross compilation is a bit complicated with nix flakes, you need to instantiate `pkgs` with hostPlatform & buildPlatform, but I'm not sure how to do it exactly

**Reply by MattSturgeon** [reply, marked as answer]
> Cross compilation is a bit complicated with nix flakes, you need to instantiate `pkgs` with hostPlatform & buildPlatform, but I'm not sure how to do it exactly 

Should be easier after https://github.com/nix-community/nixvim/issues/1784, but for now it should be possible to create a cross-comp instance of nixpkgs and pass that into nixvim.

**Comment by qlibp** [comment]
Encounter another situation, I use nix-appimage to bundle someone else nixvim flake, containg some python lsp stuff, but it doesn't work. I check the lspinfo, and it says that it need to use fuse... Is it a loop-fuse-call? since executing bundled appimage already use fuse, and when calling executable outside appimage will trigger this behavior. Any suggestion? Thanks in advance @traxys

**Reply by traxys** [reply]
I'm afraid I don't have much experience with the tool apart from knowing it exists
